884 add doc for googlecast with pillarbox (#901)
SRGSSR/pillarbox-android: pillarbox-cast/docs/README.md

# Module pillarbox-cast

> Provides a [PillarboxPlayer][ch.srgssr.pillarbox.player.PillarboxPlayer] implementation base on media 3 [CastPlayer][androidx.media3.cast.CastPlayer] that controls a Cast receiver 

Reviewer: ```suggestion
Provides a [PillarboxPlayer][ch.srgssr.pillarbox.player.PillarboxPlayer] implementation based on Media 3 [CastPlayer][androidx.media3.cast.CastPlayer] that controls a Cast receiver 
```

implementation("ch.srgssr.pillarbox:pillarbox-cast:<pillarbox_version>")
```

> The main goal of this module is to be able to build a user interface that can be use with an [PillarboxExoPlayer][ch.srgssr.pillarbox.player.PillarboxExoPlayer] or [PillarboxCastPlayer][ch.srgssr.pillarbox.cast.PillarboxCastPlayer]. Both
> implementation are base of [PillarboxPlayer][ch.srgssr.pillarbox.player.PillarboxPlayer] interface.

Reviewer: ```suggestion
The main goal of this module is to be able to build a user interface that can be used with a [PillarboxExoPlayer][ch.srgssr.pillarbox.player.PillarboxExoPlayer] or [PillarboxCastPlayer][ch.srgssr.pillarbox.cast.PillarboxCastPlayer]. Both
implementations are based on the [PillarboxPlayer][ch.srgssr.pillarbox.player.PillarboxPlayer] interface.
```

val castContext = context.getCastContext()
```

### Create the player

```kotlin
val castContext = context.getCastContext()
val player = PillarboxCastPlayer(castContext = castContext)
player.setSessionAvailabilityListener(object : SessionAvailabilityListener {
    override fun onCastSessionAvailable() {
        // The cast session is connected.
        player.setMediaItems(mediaItems)
        player.play()
        player.prepare()
    }

    override fun onCastSessionUnavailable() {
        // The cast session has been disconnected.
    }

})

> // When player is no more needed

Reviewer: ```suggestion
// When the player is not needed anymore.
```

player.release()
```

### Display a Cast button

> At some point in the application a Cast button has to be present to let the user connects to a Cast device.

Reviewer: ```suggestion
Somewhere in your application, a Cast button has to be displayed to allow the user to connect to a Cast device.
```

```kotlin
CastButton(modifier = Modifier)
```

## Local to remote playback

> With PillarboxCastPlayer it make easy to switch to local to remote and back to local playback.
> 
> When switching to remote playback, the local playback has to be stop manually and the current state of the player as to be configured to the 
> remote player.

Reviewer: ```suggestion
With `PillarboxCastPlayer`, it is easy to switch from local to remote, and back to local playback.

When switching to remote playback, the local playback has to be stop manually and the current state of the player has to be configured on the 
remote player.
```

```kotlin
val localPlayer = PillarboxExoPlayer(context)
val castContext = context.getCastContext()
val remotePlayer = PillarboxCastPlayer(castContext = castContext)
var currentPlayer: PillarboxPlayer = if (remotePlayer.isCastSessionAvailable()) remotePlayer else localPlayer
player.setSessionAvailabilityListener(object : SessionAvailabilityListener {
    override fun onCastSessionAvailable() {
        setCurrentPlayer(remotePlayer)
    }

    override fun onCastSessionUnavailable() {
        setCurrentPlayer(localPlayer)
    }
})
```

## Additional resources

- [Google Cast SDK](https://developers.google.com/cast/docs/android_sender)

[ch.srgssr.pillarbox.player.PillarboxPlayer]: https://android.pillarbox.ch/api/pillarbox-player/ch.srgssr.pillarbox.player/-pillarbox-player/index.html
[ch.srgssr.pillarbox.player.PillarboxExoPlayer]: https://android.pillarbox.ch/api/pillarbox-player/ch.srgssr.pillarbox.player/-pillarbox-exo-player.html
[ch.srgssr.pillarbox.cast.PillarboxCastPlayer]: https://android.pillarbox.ch/api/ch.srgssr.pillarbox.cast/-pillarbox-cast-player/index.html

> [androidx.media3.cast.CastPlayer]: https://github.com/androidx/media/blob/release/libraries/cast/src/main/java/androidx/media3/cast/CastPlayer.java

Reviewer: What about pointing to the documentation, instead of the source code?
```suggestion
[androidx.media3.cast.CastPlayer]: https://developer.android.com/reference/androidx/media3/cast/CastPlayer
```
